### DISCUSSION TITLE
How to Compile and Package all Dependencies

### DISCUSSION THREAD

**Discussion by isapir** [ImageMagick/ImageMagick#3655] [General]
Hello,

I need to make a portable build of ImageMagick so that I can copy and run it on another machine (same OS and arch, in this case Ubuntu 18.04 64bit) without having to install 3rd party shared libraries.  I compiled it by running

```
apt install build-essential curl git \
            libjpeg-dev libpng-dev libtiff-dev zlib1g-dev 

./configure --disable-shared \
            --enable-delegate-build \
            --prefix=/app/bin/linux/imagemagick-7
```

followed by make && make install. Then I copied the files from `/app/bin/linux/imagemagick-7` to a different server, but running `ldd` still shows that some required libraries are missing:

```
# ldd bin/magick
    linux-vdso.so.1 (0x00007ffcde3c5000)
    libjbig.so.0 => not found
    libtiff.so.5 => not found
    libjpeg.so.8 => not found
    libgomp.so.1 => not found
    /* omitted for brevity */
    /lib64/ld-linux-x86-64.so.2 (0x00007f014836c000)
```

I can copy these files to a "dependencies" directory and point to it before execution using `export LD_LIBRARY_PATH=dependencies`, but how can I compile the code so that all of the dependencies will end up in a subdirectory of the install directory?

**Comment by dlemstra** [comment]
You might want to take a look at the AppImage build of ImageMagick. You can find them here: https://github.com/ImageMagick/ImageMagick/releases. Another option would be to use a static build instead.
